### ISSUE TITLE
bufadd does not sanitize windows drive letter capitalization

### ISSUE THREAD

**Issue by mjlbach** [neovim/neovim#16331]
### Neovim version (nvim -v)

NVIM v0.6.0-dev+463-g68b2a9e56

### Vim (not Nvim) behaves the same?

N/A

### Operating system/version

Widnows 10

### Terminal name/version

Windows Termianal

### $TERM environment variable

N/A

### Installation

source build

### How to reproduce the issue

```
file_location = os.getenv('TEMP')

local file_1 = file_location .. "\\file_1.txt"
local buffer_1 = vim.fn.bufadd(file_1)
print(vim.api.nvim_buf_get_name(buffer_1))

local drive_letter = string.sub(file_location, 1, 1)
local tail = string.sub(file_location, 2)
local perturbed_path = string.lower(drive_letter) .. tail
local file_2 = perturbed_path .. "\\file_2.txt"

local buffer_2 = vim.fn.bufadd(file_2)
print(vim.api.nvim_buf_get_name(buffer_2))
```

### Expected behavior

vim.api.nvim_buf_get_name should show consistent capitalization for drive letters on windows regardless of the input to buf_add.

### Actual behavior

```
C:\Users\michael\AppData\Local\Temp\file_1.txt
c:\Users\michael\AppData\Local\Temp\file_2.txt
```

**Comment by zeertzjq** [comment]
Does this bug also apply to Vim and Nvim's `bufname()`?

**Comment by mjlbach** [comment]
yes, the behavior depends on bufadd, not buf_get_name/bufname

**Comment by erw7** [comment]
I don't think this is a neovim (or vim) bug. If `'fileignorecase'` is set properly, [`path_fnamecmp()`](https://github.com/neovim/neovim/blob/c0efe49e78eca3a17a5934cc283199122d081467/src/nvim/path.c#L331) will handle those differences properly. I think that is why `bufadd` does not do capitalization.

**Comment by mjlbach** [comment]
I guess the question is should it then. Currently when we want to compare on the lua side we have to sanitize the drive letter capitalization (which, to be fair, is trivial to do), because buf_get_name returns different values depending on the call to buff_add. I'm just not sure this behavior is intuitive

see: 
* https://github.com/neovim/nvim-lspconfig/issues/1168

**Comment by hasansujon786** [comment]
I am having the same issue. Different results for different files.
`-- c:\Users\hasan\dotfiles\nvim\lua\options.lua
 -- C:\Users\hasan\dotfiles\nvim\init.lua
`

**Comment by 3N4N** [comment]
Was there any headway into fixing this?  Does it even need fixing?  As I see it, this shouldn't cause any problem.  Sure, `bufadd()` doesn't capitalize drive letter, where is the problem in that?  It's not like when you call `buffadd()` again with the same filename, but this time with different case of drive letter, it opens a new buffer.  No, `bufadd()` behaves sensibly.  It doesn't add a new buffer, even if the second call to `buffadd()` had a different case of drive letter.

Funny aside: command prompt doesn't sanitize drive letters either.  If you do `cd /d c:\`, the prompt shows lowercase "c."  If you do `cd /d C:\`, the prompt shows uppercase "C."  No one cares.

**Comment by justinmk** [comment]
cc @ntdiary maybe this is fixed with your current PR?

**Comment by ntdiary** [comment]
> cc [@ntdiary](https://github.com/ntdiary) maybe this is fixed with your current PR?

Do you mean #39155? So far, we've only normalized path separators and haven't touched drive letter capitalization. :)
